### ISSUE TITLE
[Bug]: Every time search, querynode will crash and auto restart later

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#25357]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230627-3a222e97 
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0.dev80
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Every time search, querynode will crash and auto restart later.
```
>>> res = collection.search(vectors[:1], "float_vector", search_params, limit)
RPC error: [search], <MilvusException: (code=1, message=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=550): attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=550): context done during sleep after run#6: context deadline exceeded: fail to search on all shard leaders)>, <Time:{'RPC start': '2023-07-06 03:27:47.052590', 'RPC error': '2023-07-06 03:27:57.055696'}>
Traceback (most recent call last):
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/nico/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 682, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/root/nico/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/root/nico/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/root/nico/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/root/nico/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/root/nico/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/root/nico/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 599, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, **kwargs)
  File "/root/nico/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 579, in _execute_search_requests
    raise pre_err
  File "/root/nico/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 570, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=550): attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=549): attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_442459249473497185v0 is not available in any replica, err=NodeOffline(nodeID=550): context done during sleep after run#6: context deadline exceeded: fail to search on all shard leaders)>
```

### Expected Behavior

search successfully

### Milvus Log

devops: chaos-testing

querynodes:
```
search-it-qchgv-milvus-querycoord-676768d964-px2hk                1/1     Running                  50 (35m ago)      5h50m
search-it-qchgv-milvus-querynode-5cd5658b69-96wrs                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-bl668                 1/1     Running                  6 (4m12s ago)     11h
search-it-qchgv-milvus-querynode-5cd5658b69-c9fjt                 1/1     Running                  0                 15h
search-it-qchgv-milvus-querynode-5cd5658b69-djqv6                 1/1     Running                  0                 5h50m
search-it-qchgv-milvus-querynode-5cd5658b69-hqmvq                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-l6vvp                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-lk8f9                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-pwz4c                 1/1     Running                  0                 5h50m
search-it-qchgv-milvus-querynode-5cd5658b69-qnvx5                 1/1     Running                  6 (4m9s ago)      11h
search-it-qchgv-milvus-querynode-5cd5658b69-xp9vk                 1/1     Running                  6 (3m23s ago)     11h
```

**Comment by NicoYuan1986** [comment]
```
search-it-qchgv-milvus-querycoord-676768d964-px2hk                1/1     Running                  50 (53m ago)      6h8m
search-it-qchgv-milvus-querynode-5cd5658b69-96wrs                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-bl668                 1/1     Running                  8 (9m38s ago)     11h
search-it-qchgv-milvus-querynode-5cd5658b69-c9fjt                 1/1     Running                  0                 15h
search-it-qchgv-milvus-querynode-5cd5658b69-djqv6                 1/1     Running                  0                 6h8m
search-it-qchgv-milvus-querynode-5cd5658b69-hqmvq                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-l6vvp                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-lk8f9                 1/1     Running                  0                 11h
search-it-qchgv-milvus-querynode-5cd5658b69-pwz4c                 1/1     Running                  0                 6h8m
search-it-qchgv-milvus-querynode-5cd5658b69-qnvx5                 1/1     Running                  8 (10m ago)       11h
search-it-qchgv-milvus-querynode-5cd5658b69-xp9vk                 0/1     CrashLoopBackOff         8 (19s ago)       11h
```

**Comment by NicoYuan1986** [comment]
devops: chaos-testing

search-it-tussv-milvus-indexnode-76cbd6db95-8zx2p                 1/1     Running                  166 (9m59s ago)   35h     10.102.5.65     devops-node21   <none>           <none>
search-it-tussv-milvus-indexnode-76cbd6db95-bhbtp                 1/1     Running                  163 (12m ago)     35h     10.102.5.64     devops-node21   <none>           <none>
search-it-tussv-milvus-indexnode-76cbd6db95-fvbsm                 1/1     Running                  146 (2m12s ago)   35h     10.102.9.101    devops-node13   <none>           <none>
search-it-tussv-milvus-indexnode-76cbd6db95-gnb4b                 1/1     Running                  145 (13m ago)     35h     10.102.9.100    devops-node13   <none>           <none>
search-it-tussv-milvus-indexnode-76cbd6db95-lfmhd                 1/1     Running                  144 (10m ago)     35h     10.102.9.74     devops-node13   <none>           <none>
search-it-tussv-milvus-indexnode-76cbd6db95-rqbpf                 1/1     Running                  163 (2m49s ago)   35h     10.102.5.62     devops-node21   <none>           <none>
search-it-tussv-milvus-indexnode-76cbd6db95-t4254                 1/1     Running                  163 (7m17s ago)   35h     10.102.5.59     devops-node21   <none>           <none>
search-it-tussv-milvus-indexnode-76cbd6db95-wp22z                 1/1     Running                  156 (9m32s ago)   3d9h    10.102.9.139    devops-node13   <none>           <none>

**Comment by yah01** [comment]
This should be fixed by #25405

**Comment by yah01** [comment]
/assign @NicoYuan1986 
many fix patches merged related to this
